plugin-server/src/worker/ingestion/process-event.ts

```diff
@@ -212,16 +222,24 @@ export class EventsProcessor {
 
         const elementsChain = elements && elements.length ? elementsToString(elements) : ''
 
-        const personInfo = await this.db.getPersonData(teamId, distinctId)
         const groupProperties = await this.db.getGroupProperties(teamId, this.getGroupIdentifiers(properties))
 
+        let eventPersonUuid: string | null = null
         let eventPersonProperties: string | null = null
+        let personInfo = preIngestionEvent.person
+
         if (personInfo) {
-            // For consistency, we'd like events to contain the properties that they set, even if those were changed
-            // before the event is ingested. Thus we fetch the updated properties but override the values with the event's
-            // $set properties if they exist.
-            const latestPersonProperties = personInfo ? personInfo?.properties : {}
-            eventPersonProperties = JSON.stringify({ ...latestPersonProperties, ...(properties.$set || {}) })
+            eventPersonUuid = personInfo.uuid
+            eventPersonProperties = JSON.stringify(personInfo.properties)
+        } else {
+            personInfo = await this.db.getPersonData(teamId, distinctId)
+            if (personInfo) {
+                // For consistency, we'd like events to contain the properties that they set, even if those were changed
```

Comment:
Note: I think we can kill this clause once we move person processing before event processing